src/rest_framework_api_key/permissions.py

```diff
@@ -18,12 +18,13 @@ def get(self, request: HttpRequest) -> typing.Optional[str]:
 
     def get_from_authorization(self, request: HttpRequest) -> typing.Optional[str]:
         authorization = request.META.get("HTTP_AUTHORIZATION")
+        custom_keyword = getattr(settings, "API_KEY_CUSTOM_KEYWORD", "Api-Key")
```

Comment:
I think we should be talking about an "API key scheme", since the `Authorization` header is spec'd as being a `<Scheme>: <Value>` form (see [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes)).

What do you think?

```suggestion
        scheme = getattr(settings, "API_KEY_SCHEME", "Api-Key")
```

Other naming ideas, perhaps more explicit…

* `API_KEY_HEADER_SCHEME`
* `API_KEY_AUTHORIZATION_SCHEME`

?

Reply:
Hmm. Looking at https://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication, I see DRF's `TokenAuthentication` docs use the "keyword" wording as well.

So not fussed about either option.

But those docs make me think of a nice alternative too:

> **Note**: If you want to use a different keyword in the header, such as `Bearer`, simply subclass `TokenAuthentication` and set the `keyword` class variable.

So we could allow doing this…

```python
from rest_framework_api_key.permissions import BaseHasAPIKey, KeyParser

class BearerKeyParser(KeyParser):
    keyword = "Bearer"


class HasAPIKey(BaseHasAPIKey):
    key_parser = BearerKeyParser()

```

The nice thing about this would be that it'd allow people to do more complicated things if need be (eg multiple perm classes with different schemes and what not), as opposed to the setting option with only allows overriding the behavior once and for all (requiring you to customize everything if you need more complex behavior).

So WDYTA that `KeyParser.keyword` option?